pkg/tfshim/sdk-v2/provider2_test.go

```diff
@@ -213,6 +215,61 @@ func TestProvider2UpgradeResourceState(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "handle errors",
+			state: func() cty.Value {
+				return cty.ObjectVal(map[string]cty.Value{
+					"compute_resources": cty.ObjectVal(map[string]cty.Value{
+						"ec2_configuration": cty.ObjectVal(map[string]cty.Value{
+							"image_id_override": cty.StringVal("override"),
+						}),
+					}),
+					"id": cty.StringVal("id"),
+				})
+			}(),
+			expectErr: func(t *testing.T, err error) {
+				require.Error(t, err)
+				autogold.Expect(`1 error occurred:
+* missing expected [
```

Comment:
The error message is not a great one. 

It's losing attribute information from this String() method:

```typescript
// ValidationError wraps validation errors reported by shims (currently shim v2 and tf5)
type ValidationError struct {
	AttributePath cty.Path
	Summary       string
	Detail        string
}

func (e ValidationError) Error() string {
	if e.Detail != "" {
		return fmt.Sprintf("%s: %s", e.Summary, e.Detail)
	}
	return e.Summary
}

```

We could have had `compute_resources.$.ec2_configuration` which is essential for debugging this.

Reply:
The actual message is also terrible because the internals of tfprotov5.UpgradeResourceState handler use a raw json parser and expect types to match; no more type checking. This works out because in TF the data that's passed in is already type-conformant. Any quick ideas to improve here .